NitroxModel/DataStructures/GameLogic/Entity.cs

```diff
@@ -81,11 +136,49 @@ public Entity(Vector3 position, Quaternion rotation, Vector3 scale, TechType tec
             WaterParkGuid = waterParkGuid;
             SerializedGameObject = serializedGameObject;
             ExistsInGlobalRoot = existsInGlobalRoot;
+            ChildEntitiesByGuid = new Dictionary<string, Entity>();
+        }
+
+        [OnDeserialized]
+        private void OnDeserialize(StreamingContext streamingContext)
+        {
+            ChildEntitiesByGuid = new Dictionary<string, Entity>(); // Converts back to Dictionary
+            if (ChildEntities != null)
+            {
+                foreach (Entity child in ChildEntities)
+                {
+                    ChildEntitiesByGuid.Add(child.Guid, child);
+                }
+            }
+        }
+
+        [OnSerializing]
+        [ProtoBeforeSerialization]
+        private void OnSerialize(StreamingContext streamingContext) // Converts Dictionary to List for serialization by our friend Windows
+        {
+            List<Entity> childList = ChildEntitiesByGuid.Values.ToList();
+            if (childList == null)
+            {
+                childList = new List<Entity>();
+            }
+
+            ChildEntities = ChildEntitiesByGuid.Values.ToList();
+        }
+
+        public void AddChild(Entity child)
+        {
+            ChildEntitiesByGuid.Add(Guid, child);
         }
 
         public override string ToString()
         {
-            return "[Entity Position: " + Position + " TechType: " + TechType + " Guid: " + Guid + " Level: " + Level + " classId: " + ClassId + " ChildEntities: " + ChildEntities + " SpawnedByServer: " + SpawnedByServer + "]";
+            List<Entity> childList = ChildEntitiesByGuid.Values.ToList();
+            if (childList == null)
```

Comment:
Pretty sure `ToList()` returns an empty list (not `null`) when there are no items.

If `ChildEntitiesByGuid` were `null`, this would have already thrown a `NullReferenceException` above.